src/features/RunCode.ts

```diff
@@ -0,0 +1,86 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as path from "path";
+import vscode = require("vscode");
+import { IFeature, LanguageClient } from "../feature";
+import { SessionManager } from "../session";
+import Settings = require("../settings");
+import utils = require("../utils");
+
+enum LaunchType {
+    Debug,
+    Run,
+}
+
+export class RunCodeFeature implements IFeature {
+
+    private command: vscode.Disposable;
+    private languageClient: LanguageClient;
+
+    constructor(private sessionManager: SessionManager) {
+        this.command = vscode.commands.registerCommand(
+            "PowerShell.RunCode",
+            (runInDebugger: boolean, scriptToRun: string, args: string[]) => {
+                this.launchTask(runInDebugger, scriptToRun, args);
+            });
+    }
+
+    public dispose() {
+        this.command.dispose();
+    }
+
+    public setLanguageClient(languageClient: LanguageClient) {
+        this.languageClient = languageClient;
+    }
+
+    private async launchTask(
+        runInDebugger: boolean,
+        scriptToRun: string,
+        args: string[]) {
+
+        const launchType = runInDebugger ? LaunchType.Debug : LaunchType.Run;
+        const launchConfig = this.createLaunchConfig(launchType, scriptToRun, args);
+        this.launch(launchConfig);
+    }
+
+    private createLaunchConfig(launchType: LaunchType, scriptToRun: string, args: string[]) {
+        const currentDocument = vscode.window.activeTextEditor.document;
+        const settings = Settings.load();
+
+        // Since we pass the script path to PSES in single quotes to avoid issues with PowerShell
```

Comment:
Should the script path be a URI instead maybe?

Reply:
I should update that... it should really be "commandToRun" but in a launch config, we call it a "script"

Reply:
it shouldn't be a URI though, because `scriptToRun` could be `Invoke-Build` with an `args` of `Clean`

Reply:
Understood. Tricky to know what the right solution is when a path is part of a script.

In that case, is there a reason we need to quote-escape the string here rather than on the PSES side? We definitely already have functionality for that server side.

Reply:
Just trying to sort out where our "layers" should be for paths and scripts going from user input through the client to the server and then executed

Reply:
@rjmholt there actually already _is_ quote-escaping on the PSES side 😄this is what gets hit:

https://github.com/PowerShell/PowerShellEditorServices/blob/02d21f79c615b78a9d9d627826a8d381b74b5b57/src/PowerShellEditorServices/Session/PowerShellContext.cs#L964-L979

Reply:
Yeah that's what I was thinking!

The comment makes it sound like we're doing it on this side as well though. Is it just out of date?

Reply:
Yeah the comment (stolen from the PesterTests feature) was actually for an _argument_ not the _script_ lol

https://github.com/PowerShell/vscode-powershell/blob/master/src/features/PesterTests.ts#L79

So it made sense in that context... but can be deleted here.

Reply:
Ah makes sense